core/lib/assert.c

```diff
@@ -24,29 +24,34 @@
 #include "backtrace.h"
 #endif
 
-__NORETURN void _assert_failure(const char *file, unsigned line)
+__NORETURN static inline void _assert_common(void)
 {
-    printf("%s:%u => ", file, line);
 #if IS_USED(MODULE_BACKTRACE)
+#ifdef DEBUG_ASSERT_VERBOSE
     printf("failed assertion. Backtrace:\n");
+#endif
     backtrace_print();
 #endif
 #ifdef DEBUG_ASSERT_BREAKPOINT
     DEBUG_BREAKPOINT(1);
 #endif
+    if (DEBUG_ASSERT_ZOMBIFY) {
+        puts("FAILED ASSERTION.");
+        thread_zombify();
```

Comment:
When in an ISR, this is a no-op, but the zombify does not state what happens when IRQs are off. Does that need special handling? (In #20626 my assumption was that we'd also have to crash then, because we can never enable IRQs, only restore them).

Reply:
We can call `irq_enable()`

Reply:
That is a very dangerous function that can easily cause unintended behavior. If IRQs are off, I think we'll need to crash the system just as if we were in an interrupt. (I'd say "see disucssion at" https://github.com/RIOT-OS/RIOT/pull/17433, but that was not so much discussion as just ACK'd).